### ISSUE TITLE
Error when specifying slug as a read-only field

### ISSUE THREAD

**Issue by KupeRr** [wagtail/wagtail#11442]
### Issue Summary

I had a goal, to limit the user's ability to change slug on specific pages.
However, I encountered a problem that I found quite difficult to analyze or understand how I could affect it

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. Create a simple page class that inherits from wagtailcore.models.Page
3. Specify slug field as read-only field, like `promote_panels = [FieldPanel("slug", read_only=True)]`
4. Open the page for editing through the administration panel

I expected the slug field to be displayed just like any other read-only field in the admin panel, I was referring to the information from version 5.1

### Technical details

-   Python version: 3.11.7.
-   Django version: 4.2.8.
-   Wagtail version: 5.2.1.
-   Browser version: Chrome 120.

**Comment by gasman** [comment]
Thanks for the report @KupeRr! When reporting an issue, please remember to include a description of what exactly happens - simply saying "I encountered a problem" or "there is an error" isn't very helpful for understanding the issue.

I tried it on a new project and got this error:
<details>

```
Environment:


Request Method: GET
Request URL: http://localhost:8000/admin/pages/3/edit/

Django Version: 5.0
Python Version: 3.11.1
Installed Applications:
['home',
 'search',
 'wagtail.contrib.forms',
 'wagtail.contrib.redirects',
 'wagtail.embeds',
 'wagtail.sites',
 'wagtail.users',
 'wagtail.snippets',
 'wagtail.documents',
 'wagtail.images',
 'wagtail.search',
 'wagtail.admin',
 'wagtail',
 'modelcluster',
 'taggit',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles']
Installed Middleware:
['django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'wagtail.contrib.redirects.middleware.RedirectMiddleware']


Template error:
In template /Users/matthew/Development/tbx/wagtail/devscript/wagtail/wagtail/admin/templates/wagtailadmin/panels/object_list.html, error at line 9
   Key 'slug' not found in 'HomePageForm'. Choices are: comment_notifications, expire_at, go_live_at, title.
   1 : {% load wagtailadmin_tags %}
   2 : 
   3 : <div class="w-form-width" {% include "wagtailadmin/shared/attrs.html" with attrs=self.attrs %}>
   4 :     {% if self.help_text %}
   5 :         {% help_block status="info" %}{{ self.help_text }}{% endhelp_block %}
   6 :     {% endif %}
   7 :     {% for child, identifier in self.visible_children_with_identifiers %}
   8 :         {% panel id_prefix=self.prefix id=identifier classname=child.classes|join:' ' attrs=child.attrs heading=child.heading heading_size="label" icon=child.icon id_for_label=child.id_for_label is_required=child.is_required %}
   9 :              {% component child %} 
   10 :         {% endpanel %}
   11 :     {% endfor %}
   12 : </div>
   13 : 

Traceback (most recent call last):
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/forms/forms.py", line 174, in __getitem__
    field = self.fields[name]
            ^^^^^^^^^^^^^^^^^

During handling of the above exception ('slug'), another exception occurred:
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/core/handlers/base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "/Users/matthew/Development/tbx/wagtail/devscript/wagtail/wagtail/admin/auth.py", line 169, in overridden_render
    return render()
           ^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1059, in render
    output = self.filter_expression.resolve(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 909, in _resolve_lookup
    current = current()
              ^^^^^^^^^
  File "/Users/matthew/Development/tbx/wagtail/devscript/wagtail/wagtail/admin/panels/base.py", line 317, in render_form_content
    return mark_safe(self.render_html() + self.render_missing_fields())
                     ^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/laces/components.py", line 40, in render_html
    return template.render(context_data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/defaulttags.py", line 241, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/defaulttags.py", line 325, in render
    return nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1059, in render
    output = self.filter_expression.resolve(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 909, in _resolve_lookup
    current = current()
              ^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/laces/components.py", line 40, in render_html
    return template.render(context_data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/defaulttags.py", line 241, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/Development/tbx/wagtail/devscript/wagtail/wagtail/admin/templatetags/wagtailadmin_tags.py", line 1036, in render
    children = self.nodelist.render(context) if self.nodelist else ""
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/laces/templatetags/laces.py", line 54, in render
    html = component.render_html(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/laces/components.py", line 35, in render_html
    context_data = self.get_context_data(parent_context)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/Development/tbx/wagtail/devscript/wagtail/wagtail/admin/panels/title_field_panel.py", line 59, in get_context_data
    field.widget.attrs.update(**self.get_attrs())
                                ^^^^^^^^^^^^^^^^
  File "/Users/matthew/Development/tbx/wagtail/devscript/wagtail/wagtail/admin/panels/title_field_panel.py", line 80, in get_attrs
    targets = [self.get_target_selector(target) for target in panel.targets]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/Development/tbx/wagtail/devscript/wagtail/wagtail/admin/panels/title_field_panel.py", line 80, in <listcomp>
    targets = [self.get_target_selector(target) for target in panel.targets]
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/Development/tbx/wagtail/devscript/wagtail/wagtail/admin/panels/title_field_panel.py", line 127, in get_target_selector
    field = self.form[target]
            ^^^^^^^^^^^^^^^^^
  File "/Users/matthew/.virtualenvs/wagtail/lib/python3.11/site-packages/django/forms/forms.py", line 176, in __getitem__
    raise KeyError(
    ^

Exception Type: KeyError at /admin/pages/3/edit/
Exception Value: "Key 'slug' not found in 'HomePageForm'. Choices are: comment_notifications, expire_at, go_live_at, title."
```
</details>

This is because TitleFieldPanel (as used for the `title` field) tries to retrieve the ID of the slug field to set up the auto-filling behaviour, and fails because this is no longer part of the form. I think it would be reasonable for TitleFieldPanel to silently handle this situation, and disable the auto-filling behaviour in this case. It's possible that fixing this will reveal further errors, though, and I'm not 100% sure that making the slug read-only is something we should support.

**Comment by salvo-polizzi** [comment]
Hi, I would like to work on this issue.  Can someone help me understand how to disable the auto-filling behavior for the slug?

> Thanks for the report @KupeRr! When reporting an issue, please remember to include a description of what exactly happens - simply saying "I encountered a problem" or "there is an error" isn't very helpful for understanding the issue.
> 
> I tried it on a new project and got this error:
> 
> This is because TitleFieldPanel (as used for the `title` field) tries to retrieve the ID of the slug field to set up the auto-filling behaviour, and fails because this is no longer part of the form. I think it would be reasonable for TitleFieldPanel to silently handle this situation, and disable the auto-filling behaviour in this case. It's possible that fixing this will reveal further errors, though, and I'm not 100% sure that making the slug read-only is something we should support.

**Comment by gasman** [comment]
Thanks @salvo-polizzi! The link between the title and slug field is set up by the `data-` attributes added here:

https://github.com/wagtail/wagtail/blob/f0a67975db817a54a7c685ec618ee146b89ee839/wagtail/admin/panels/title_field_panel.py#L73-L81

so I think you'll want to omit some or all of those. Maybe @lb- has some more pointers...

**Comment by laymonage** [comment]
It should be pretty straightforward to add `if target in self.form` to this list comprehension:
https://github.com/wagtail/wagtail/blob/f0a67975db817a54a7c685ec618ee146b89ee839/wagtail/admin/panels/title_field_panel.py#L80

I tried it and it works – just needs tests!

**Comment by salvo-polizzi** [comment]
> It should be pretty straightforward to add `if target in self.form` to this list comprehension:
> 
> https://github.com/wagtail/wagtail/blob/f0a67975db817a54a7c685ec618ee146b89ee839/wagtail/admin/panels/title_field_panel.py#L80
> 
> I tried it and it works – just needs tests!
> 
> <img alt="image" width="194" src="https://private-user-images.githubusercontent.com/6379424/295982897-1cea0b50-cf7d-426c-ad2a-89130ba144a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ5OTA4NjYsIm5iZiI6MTcwNDk5MDU2NiwicGF0aCI6Ii82Mzc5NDI0LzI5NTk4Mjg5Ny0xY2VhMGI1MC1jZjdkLTQyNmMtYWQyYS04OTEzMGJhMTQ0YTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDExMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMTFUMTYyOTI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2ZkYzU5ZjE1N2M4Y2YxMzRhNzBhY2M3ZDY5MzAxNjI2YmNmMjNkNDc0Njc2MWNlY2ExNWI2NDZhNDRiZTIzZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.KkSVMfiwX6cumKb4SKMybwvIJVo80KtqanUlRHtMeTw">

ok, got it

**Comment by rohitsrma** [comment]
I'm Sorry @salvo-polizzi, for jumping ahead of you on this one. Actually i was already working on it just forgot to add a comment.

And, thanks @laymonage for your suggestion. it worked like a charm. I was trying to remove the "slug" from the **Targets** list if read-only is true. However, your suggested approach looks perfect.

**Comment by gasman** [comment]
Fixed in #11447
